test: Treat `bitcoin-wallet` binary in the same way as others

hebasto opened the pull request:
This PR makes the `bitcoin-wallet` binary path customizable in the same way how it can be done now with other ones, including `bitcoind`, `bitcoin-cli` and `bitcoin-util`.

hebasto commented on test/functional/test_framework/test_framework.py:
```diff
@@ -252,9 +252,15 @@ def setup(self):
         self.options.bitcoind = os.getenv("BITCOIND", default=fname_bitcoind)
         self.options.bitcoincli = os.getenv("BITCOINCLI", default=fname_bitcoincli)
         self.options.bitcoinutil = os.getenv("BITCOINUTIL", default=fname_bitcoinutil)
+        self.options.bitcoinwallet = os.getenv("BITCOINWALLET", default=fname_bitcoinwallet)
```
> What about making this a function to reduce code bloat while touching this?

Done.

maflcko reviewed (approved):
lgtm

stickies-v reviewed (approved):
ACK ef01337bbb3d5269db6a26e42c7bbd238000ce2f

Nice little refactoring (modulo being able to now set `BITCOINWALLET` which is new behaviour), less code duplication and functions with smaller scope.

hebasto commented:
Updated ef01337bbb3d5269db6a26e42c7bbd238000ce2f -> de8b7ab2e023e28168e870629c2423dab589085e ([pr27554.03](https://github.com/hebasto/bitcoin/commits/pr27554.03) -> [pr27554.04](https://github.com/hebasto/bitcoin/commits/pr27554.04), [diff](https://github.com/hebasto/bitcoin/compare/pr27554.03..pr27554.04)):

- addressed @stickies-v's comments

stickies-v commented on test/functional/test_framework/test_framework.py:
```diff
@@ -228,6 +228,22 @@ def parse_args(self):
 
         PortSeed.n = self.options.port_seed
 
+    def set_binary_paths(self):
```
Oops sorry, the docstring is meant to go underneath the function `def`, not above. My bad.

hebasto replied:
Adjusted.

hebasto committed:
test, refactor: Add `set_binary_paths` function

This change factors out the repeated code into a new `set_binary_paths`
function.

hebasto committed:
test: Treat `bitcoin-wallet` binary in the same way as others

This change makes the `bitcoin-wallet` binary path customizable in the
same way how it can be done now with other ones, including `bitcoind`,
`bitcoin-cli` and `bitcoin-util`.

stickies-v reviewed (approved):
re-ACK f6d7636be4eb0b19878428906dd5e394df7d07a2
